uReplicator-Controller/src/main/java/com/uber/stream/kafka/mirrormaker/controller/core/OffsetMonitor.java

```diff
@@ -134,15 +119,42 @@ public void start() {
       int delaySec = 60 + new Random().nextInt(240);
       logger.info("OffsetMonitor starts updating offsets every {} seconds with delay {} seconds", refreshIntervalInSec,
           delaySec);
-      logger.info("OffsetMonitor starts with brokerList=" + srcBrokerList);
+
+      MetricRegistry metricRegistry = HelixKafkaMirrorMakerMetricsReporter.get().getRegistry();
+      Counter counter = metricRegistry.counter("offsetMonitor.executed");
 
       refreshExecutor.scheduleAtFixedRate(new Runnable() {
         @Override
         public void run() {
           logger.info("TopicList starts updating");
+          if (zkClientQueue == null) {
+
+            zkClientQueue = new LinkedBlockingQueue<>(numOffsetThread);
+            for (int i = 0; i < numOffsetThread; i++) {
+              ZkClient zkClient = new ZkClient(offsetZkString, 30000, 30000, ZKStringSerializer$.MODULE$);
+              zkClientQueue.add(zkClient);
+            }
+
+            ZkClient zkClient = new ZkClient(srcZkString, 30000, 30000, ZKStringSerializer$.MODULE$);
+            List<String> brokerIdList = zkClient.getChildren("/brokers/ids");
+            JSONParser parser = new JSONParser();
+
+            // Hacky solution for now to reduce init time
+            int count = 0;
```

Comment:
remove unused parameter count and the comment since the hacky solution not exists anymore.